### PR TITLE
feat(sample-function-aws): adding aws greeting and aws book functions

### PR DESCRIPTION
**Pull request by whoswendy**
This PR contains code to fix the aws function sample

### CODE REVIEW

File: function/function-samples/sample-function-book-aws/src/main/java/io/americanexpress/function/book/config/BookConfig.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+ * the License.
+ */
+package io.americanexpress.function.book.config;
+
+import io.americanexpress.data.book.config.BookDataConfig;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Import;
+
+@Configuration
+@Import({BookDataConfig.class})
```
Comment (guttff) [inline function/function-samples/sample-function-book-aws/src/main/java/io/americanexpress/function/book/config/BookConfig.java]:
```suggestion
@Import(BookDataConfig.class)
```

File: function/function-samples/sample-function-book-aws/src/main/java/io/americanexpress/function/book/function/BookFunction.java [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+import org.springframework.web.server.ResponseStatusException;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+@Configuration
+public class BookFunction {
```
Comment (guttff) [inline function/function-samples/sample-function-book-aws/src/main/java/io/americanexpress/function/book/function/BookFunction.java:42]: add javadoc

File: function/function-samples/sample-function-book-aws/src/main/java/io/americanexpress/function/book/function/BookFunction.java [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+    private final String BOOK_NOT_FOUND = "Book Not Found";
+
+    private final BookRepository bookRepository;
```
Comment (guttff) [inline function/function-samples/sample-function-book-aws/src/main/java/io/americanexpress/function/book/function/BookFunction.java:49]: add javadoc

File: function/function-samples/sample-function-book-aws/src/main/java/io/americanexpress/function/book/function/BookFunction.java [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+    public BookFunction(BookRepository bookRepository) {
```
Comment (guttff) [inline function/function-samples/sample-function-book-aws/src/main/java/io/americanexpress/function/book/function/BookFunction.java:56]: add javadoc

File: function/function-samples/sample-function-book-aws/src/main/java/io/americanexpress/function/book/function/BookFunction.java [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+        this.bookRepository = bookRepository;
+    }
+
+    @Bean
```
Comment (guttff) [inline function/function-samples/sample-function-book-aws/src/main/java/io/americanexpress/function/book/function/BookFunction.java:65]: add javadoc

File: function/function-samples/sample-function-book-aws/src/main/java/io/americanexpress/function/book/function/BookFunction.java [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+    public Function<Message<CreateBookRequest>, Mono<CreateBookResponse>> create() {
+        return request -> bookRepository.save(BookEntityCreator.create(request.getPayload().getTitle(), request.getPayload().getAuthor(), 1))
+                    .map(bookEntity -> new CreateBookResponse());
+    }
+
+    @Bean
```
Comment (guttff) [inline function/function-samples/sample-function-book-aws/src/main/java/io/americanexpress/function/book/function/BookFunction.java:76]: add javadoc

File: function/function-samples/sample-function-book-aws/src/main/java/io/americanexpress/function/book/function/BookFunction.java [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+    public Consumer<Message<CreateBookRequest>> createBook() {
+        return request -> bookRepository.save(BookEntityCreator.create(request.getPayload().getTitle(), request.getPayload().getAuthor(), 1));
+    }
+
+    @Bean
```
Comment (guttff) [inline function/function-samples/sample-function-book-aws/src/main/java/io/americanexpress/function/book/function/BookFunction.java:86]: add javadoc

File: function/function-samples/sample-function-book-aws/src/main/java/io/americanexpress/function/book/function/BookFunction.java [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+    public Function<Message<ReadBookRequest>, Mono<ReadBookResponse>> read() {
+        return request -> bookRepository.findByTitleAndAuthor(request.getPayload().getTitle(), request.getPayload().getAuthor())
+                .map(ReadBookResponseCreator::create);
+    }
+
+    @Bean
```
Comment (guttff) [inline function/function-samples/sample-function-book-aws/src/main/java/io/americanexpress/function/book/function/BookFunction.java:97]: add javadoc

File: function/function-samples/sample-function-book-aws/src/main/java/io/americanexpress/function/book/function/BookFunction.java [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+    public Supplier<Flux<ReadBookResponse>> get() {
+        return () -> bookRepository.findAll().map(ReadBookResponseCreator::create);
+    }
+
+    @Bean
```
Comment (guttff) [inline function/function-samples/sample-function-book-aws/src/main/java/io/americanexpress/function/book/function/BookFunction.java:107]: add javadoc

File: function/function-samples/sample-function-book-aws/src/main/java/io/americanexpress/function/book/function/BookFunction.java [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+    public Consumer<Message<UpdateBookRequest>> update() {
+        return request -> bookRepository.findByTitleAndAuthor(request.getPayload().getTitle(), request.getPayload().getAuthor())
+                .switchIfEmpty(Mono.error(new ResponseStatusException(HttpStatus.NOT_FOUND, BOOK_NOT_FOUND)))
+                .map(book -> updateBook(book, request.getPayload().getNumberOfCopies()))
+                .flatMap(bookRepository::save);
+    }
+
+    @Bean
```
Comment (guttff) [inline function/function-samples/sample-function-book-aws/src/main/java/io/americanexpress/function/book/function/BookFunction.java:120]: add javadoc

File: function/function-samples/sample-function-book-aws/src/main/java/io/americanexpress/function/book/function/BookFunction.java [hunk 11]
```diff
@@ -0,0 +1,100 @@
...
+    public Function<Message<UpdateBookRequest>, Mono<CreateBookResponse>> updateBook() {
+        return request -> bookRepository.findByTitleAndAuthor(request.getPayload().getTitle(), request.getPayload().getAuthor())
+                .switchIfEmpty(Mono.error(new ResponseStatusException(HttpStatus.NOT_FOUND, BOOK_NOT_FOUND)))
+                .map(book -> updateBook(book, request.getPayload().getNumberOfCopies()))
+                .flatMap(bookRepository::save)
+                .map(bookEntity -> new CreateBookResponse());
+    }
+
+    private BookEntity updateBook(BookEntity book, int numOfCopies){
```
Comment (guttff) [inline function/function-samples/sample-function-book-aws/src/main/java/io/americanexpress/function/book/function/BookFunction.java:136]: add javadoc

File: function/function-samples/sample-function-book-aws/src/main/java/io/americanexpress/function/book/function/BookFunction.java [hunk 12]
```diff
@@ -0,0 +1,100 @@
...
+        if (book != null) book.setNumberOfCopies(numOfCopies);
+        return book;
+    }
+
+    @Bean
```
Comment (guttff) [inline function/function-samples/sample-function-book-aws/src/main/java/io/americanexpress/function/book/function/BookFunction.java:146]: add javadoc

File: function/function-samples/sample-function-book-aws/src/main/java/io/americanexpress/function/book/function/helper/ReadBookResponseCreator.java [hunk 13]
```diff
@@ -0,0 +1,42 @@
...
+
+import io.americanexpress.data.book.entity.BookEntity;
+import io.americanexpress.function.book.model.ReadBookResponse;
+
+/**
+ * {@code ReadBookResponseCreator} is the helper class for creating a {@link ReadBookResponse}.
+ */
+public class ReadBookResponseCreator {
+
+    private ReadBookResponseCreator() {}
```
Comment (guttff) [inline function/function-samples/sample-function-book-aws/src/main/java/io/americanexpress/function/book/function/helper/ReadBookResponseCreator.java:27]: add javadoc

File: function/function-samples/sample-function-greeter-aws/pom.xml [hunk 14]
```diff
@@ -31,6 +31,16 @@
...
+        <dependency>
+            <groupId>org.springframework.cloud</groupId>
+            <artifactId>spring-cloud-starter-function-web</artifactId>
+            <version>3.2.7</version>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework.cloud</groupId>
+            <artifactId>spring-cloud-function-adapter-aws</artifactId>
+            <version>3.2.7</version>
+        </dependency>
```
Comment (guttff) [inline function/function-samples/sample-function-greeter-aws/pom.xml]:
```suggestion
        <dependency>
            <groupId>org.springframework.cloud</groupId>
            <artifactId>spring-cloud-function-adapter-aws</artifactId>
            <version>3.2.7</version>
        </dependency>
         <dependency>
            <groupId>org.springframework.cloud</groupId>
            <artifactId>spring-cloud-starter-function-web</artifactId>
            <version>3.2.7</version>
        </dependency>
```

Comment (guttff) [reply function/function-samples/sample-function-greeter-aws/pom.xml]: can we add ${version} for this?

File: function/function-samples/sample-function-greeter-aws/src/main/java/io/americanexpress/function/greeting/rest/CloudFunctionMain.java [hunk 15]
```diff
@@ -0,0 +1,41 @@
...
+import java.util.function.Function;
+
+@SpringBootApplication
+public class CloudFunctionMain {
+
+    public static void main(String[] args) {
+        SpringApplication.run(CloudFunctionMain.class, args);
+    }
+
+//    @Bean("toUpperCase")
```
Comment (guttff) [inline function/function-samples/sample-function-greeter-aws/src/main/java/io/americanexpress/function/greeting/rest/CloudFunctionMain.java]: remove comment

Comment (guttff) [reply function/function-samples/sample-function-greeter-aws/src/main/java/io/americanexpress/function/greeting/rest/CloudFunctionMain.java]: add javadoc

File: function/function-samples/sample-function-greeter-aws/src/main/java/io/americanexpress/function/greeting/rest/CloudFunctionMain.java [hunk 16]
```diff
@@ -0,0 +1,41 @@
...
+    @Bean
+    public Function<String, String> uppercase() {
+        return value -> value.toUpperCase();
+    }
+
+
+    @Bean
```
Comment (guttff) [inline function/function-samples/sample-function-greeter-aws/src/main/java/io/americanexpress/function/greeting/rest/CloudFunctionMain.java]: add javadoc

File: function/function-samples/sample-function-greeter-aws/src/main/java/io/americanexpress/function/greeting/rest/config/Config.java [hunk 17]
```diff
@@ -0,0 +1,28 @@
...
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.validation.Validator;
+import org.springframework.validation.beanvalidation.LocalValidatorFactoryBean;
+
+@Configuration
+public class Config {
+
+    @Bean
+    public Validator springValidator() {
```
Comment (guttff) [inline function/function-samples/sample-function-greeter-aws/src/main/java/io/americanexpress/function/greeting/rest/config/Config.java]: add javadoc

File: function/function-samples/sample-function-greeter-aws/src/main/java/io/americanexpress/function/greeting/rest/config/Config.java [hunk 18]
```diff
@@ -0,0 +1,28 @@
...
+ * the License.
+ */
+package io.americanexpress.function.greeting.rest.config;
+
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.validation.Validator;
+import org.springframework.validation.beanvalidation.LocalValidatorFactoryBean;
+
+@Configuration
```
Comment (guttff) [inline function/function-samples/sample-function-greeter-aws/src/main/java/io/americanexpress/function/greeting/rest/config/Config.java]: add javadoc

File: function/function-samples/sample-function-greeter-aws/src/main/java/io/americanexpress/function/greeting/rest/function/Greet.java [hunk 19]
```diff
@@ -0,0 +1,49 @@
...
+import org.springframework.web.server.ResponseStatusException;
+
+import java.util.function.Function;
+
+public class Greet implements Function<Greeting, Greeting> {
+
+    @Autowired
+    protected Validator validator;
+
+    @Override
```
Comment (guttff) [inline function/function-samples/sample-function-greeter-aws/src/main/java/io/americanexpress/function/greeting/rest/function/Greet.java:44]: add javadoc

File: function/function-samples/sample-function-greeter-aws/src/main/java/io/americanexpress/function/greeting/rest/function/Greet.java [hunk 20]
```diff
@@ -0,0 +1,49 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.validation.BeanPropertyBindingResult;
+import org.springframework.validation.Errors;
+import org.springframework.validation.Validator;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.util.function.Function;
+
+public class Greet implements Function<Greeting, Greeting> {
```
Comment (guttff) [inline function/function-samples/sample-function-greeter-aws/src/main/java/io/americanexpress/function/greeting/rest/function/Greet.java:28]: add javadoc

File: function/function-samples/sample-function-greeter-aws/src/main/java/io/americanexpress/function/greeting/rest/function/Greeter.java [hunk 21]
```diff
@@ -0,0 +1,24 @@
...
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.americanexpress.function.greeting.rest.function;
+
+import java.util.function.Function;
+
+public class Greeter implements Function<String, String> {
+
+    @Override
```
Comment (guttff) [inline function/function-samples/sample-function-greeter-aws/src/main/java/io/americanexpress/function/greeting/rest/function/Greeter.java:23]: add javadoc

File: function/function-samples/sample-function-greeter-gcp/src/main/java/io/americanexpress/function/greeting/rest/CloudFunctionMain.java [hunk 22]
```diff
@@ -14,7 +27,7 @@ public static void main(String[] args) {
     }
 
     @Bean
-    public Function<String, String> uppercase_hello() {
+    public Function<String, String> uppercase() {
```
Comment (guttff) [inline function/function-samples/sample-function-greeter-gcp/src/main/java/io/americanexpress/function/greeting/rest/CloudFunctionMain.java:30]: add javadoc

File: function/function-samples/sample-function-greeter-aws/src/main/java/io/americanexpress/function/greeting/rest/model/Greeting.java [hunk 23]
```diff
@@ -0,0 +1,30 @@
...
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.americanexpress.function.greeting.rest.model;
+
+import javax.validation.constraints.NotBlank;
+
+public class Greeting {
```
Comment (guttff) [inline function/function-samples/sample-function-greeter-aws/src/main/java/io/americanexpress/function/greeting/rest/model/Greeting.java:18]: add javadoc

File: function/function-samples/sample-function-greeter-aws/src/main/java/io/americanexpress/function/greeting/rest/config/Config.java [hunk 24]
```diff
@@ -0,0 +1,28 @@
...
+public class Config {
```
Comment (guttff) [inline function/function-samples/sample-function-greeter-aws/src/main/java/io/americanexpress/function/greeting/rest/config/Config.java]: Rename the file GreeterConfig

File: function/function-samples/sample-function-greeter-aws/src/main/java/io/americanexpress/function/greeting/rest/CloudFunctionMain.java [hunk 25]
```diff
@@ -0,0 +1,41 @@
...
+package io.americanexpress.function.greeting.rest;
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.context.annotation.Bean;
+
+import java.util.function.Function;
+
+@SpringBootApplication
+public class CloudFunctionMain {
```
Comment (guttff) [inline function/function-samples/sample-function-greeter-aws/src/main/java/io/americanexpress/function/greeting/rest/CloudFunctionMain.java]: rename this GreeterApplication

**Review by andCastStatic — APPROVED** [review]

**Review by guttff — APPROVED** [review]

**Pull request merged**